python/spark_package/spark_package.py

```diff
@@ -446,8 +449,7 @@ def validate_and_return_sp_dep(line):
 
 
 def pom_pretty_print(f):
-    return '\n'.join([line for line in dom.parseString(f)
-        .toprettyxml(indent=' ' * 2, encoding='UTF-8').split('\n') if line.strip()])
+    return dom.parseString(f).toprettyxml(indent=' ' * 2, encoding='UTF-8').decode("utf-8").strip('\n')
```

Comment:
FWIW, it seems that `toprettyxml` under python2 involves silly whitespaces.
http://ronrothman.com/public/leftbraned/xml-dom-minidom-toprettyxml-and-silly-whitespace/#best-solution